cypress/integration/core-opensearch-dashboards/opensearch-dashboards/datasource-management-plugin/2_datasource_table.spec.js

```diff
@@ -0,0 +1,450 @@
+          'exist'
+        );
+      });
+    });
+
+    describe('Sorting', () => {
+      before(() => {
+        // Create 25+ data sources that can be sorted alphabetically by using letters a-z
+        for (let i = 97; i < 123; i++) {
+          const char = String.fromCharCode(i);
+          const dataSourceJSON = {
+            attributes: {
+              title: `ds_${char}`,
+              description: `test ds_description_${char}`,
+              endpoint: `https://dummy${char}`,
+              auth: {
+                type: i % 2 ? 'username_password' : 'no_auth',
+              },
+            },
+          };
+          if (dataSourceJSON.attributes.auth.type === 'username_password') {
+            dataSourceJSON.attributes.auth.credentials = {
+              username: char,
+              password: char,
+            };
+          }
+          cy.createDataSource(dataSourceJSON);
+        }
+      });
+
+      // Sort by title - by default
+      it('should be sorted in ascending order on Datasource title column by default', () => {
+        // Confirm we have ds_a in view and not ds_z
+        cy.contains('ds_a').should('exist');
+        cy.contains('ds_z').should('not.exist');
+      });
+
+      // sort by description
+      it('should be sorted in ascending/descending order on description column', () => {
+        // case 1: Ascending order
+        // Get the datasource table header and click it to sort ascending first
+        cy.get('thead > tr > th')
+          .contains('Description')
+          .click({ force: true });
+        // Confirm we have "test ds_description_a" in view and not "test ds_description_z"
+        cy.contains('test ds_description_a').should('exist');
+        cy.contains('test ds_description_z').should('not.exist');
+
+        // case 2: Descending order
+        // Get the datasource table header and click it to sort descending
+        cy.get('thead > tr > th')
+          .contains('Description')
+          .click({ force: true });
+        // Confirm we have "test ds_description_a" in view and not "test ds_description_z"
+        cy.contains('test ds_description_a').should('not.exist');
+        cy.contains('test ds_description_z').should('exist');
+      });
+
+      // sort descending by data source title
+      it('should be sorted in descending order by Datasource title', () => {
+        // Get the datasource table header and click it to sort descending first
+        cy.get('thead > tr > th').contains('Datasource').click({ force: true });
+
+        // Confirm we have ds_z in view and not ds_a
+        cy.contains('ds_z');
+        cy.contains('ds_a').should('not.exist');
+      });
+    });
+
+    describe('Search', () => {
+      // case 1: single match
+      it('should be able to search for single match', () => {
+        // Clear & Type in ds_z in search input
+        cy.get(`input[type="search"]`).focus().clear();
```

Comment:
nit: we usually use single quotes since there are no variables inside the string